src/sentry/api/event_search.py

```diff
@@ -1188,4 +1195,65 @@ def parse_search_query(
 
     if config_overrides:
         config = SearchConfig.create_from(config, **config_overrides)
+
     return SearchVisitor(config, params=params, builder=builder).visit(tree)
+
+
+def cleanup_search_query(tokens: Sequence[QueryToken]) -> Sequence[QueryToken]:
```

Comment:
Moved this function here, since it's specific to the search query and will be shared between on demand extraction and custom rules generation.